File: csharp practice/Jukebox/Pop.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace JukeboxCS
{
    public class Pop : Song
    {
        public Pop(string mytitle, string myartist)
        {
            title = mytitle;
            artist = myartist;
            genre = "Pop";
        }

        public override bool Rate(int myrate)
        {
            if (myrate > 0 && myrate < 5)
            {
                ++ratecount;
                ratesum += myrate;
                return true;
            }
            else
            {
                return false;
            }
        }

    }
}

File: week-05/Jukebox/Pop.cpp
#include <string>
#include "Pop.h"

using namespace std;

Pop::Pop(string title, string artist) : Song (title, artist) {
    genre = "Pop";
}

bool Pop::rate(unsigned int _rate) {
    if (_rate > 1 && _rate < 6) {
        ++count_of_ratings;
        sum_of_ratings += _rate;
        return true;
    } else {
        return false;
    }
}